paasta_tools/kubernetes_tools.py

```diff
@@ -320,6 +321,7 @@ class KubernetesDeploymentConfigDict(LongRunningServiceConfigDict, total=False):
     prometheus_port: int
     routable_ip: bool
     pod_management_policy: str
+    sidecar_injection: bool
```

Comment:
should we have a more explicit name here just in case this flag hangs around for a while? wouldn't want us to have to inject another (unrelated) sidecar in the future and have this flag still around :p